ethportal-api/src/types/consensus/kzg_commitment.rs

```diff
@@ -0,0 +1,138 @@
+#[ssz(struct_behaviour = "transparent")]
+pub struct KzgCommitment(pub [u8; BYTES_PER_COMMITMENT]);
+
+impl KzgCommitment {
+    pub fn calculate_versioned_hash(&self) -> Hash256 {
+        let mut versioned_hash = hash_fixed(&self.0);
+        versioned_hash[0] = VERSIONED_HASH_VERSION_KZG;
+        Hash256::from_slice(versioned_hash.as_slice())
+    }
+
+    pub fn empty_for_testing() -> Self {
+        KzgCommitment([0; c_kzg::BYTES_PER_COMMITMENT])
+    }
+}
+
+impl From<KzgCommitment> for c_kzg::Bytes48 {
+    fn from(value: KzgCommitment) -> Self {
+        value.0.into()
+    }
+}
+
+impl Display for KzgCommitment {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "0x")?;
+        for i in &self.0[0..2] {
+            write!(f, "{i:02x}")?;
+        }
+        write!(f, "…")?;
+        for i in &self.0[BYTES_PER_COMMITMENT - 2..BYTES_PER_COMMITMENT] {
+            write!(f, "{i:02x}")?;
+        }
+        Ok(())
+    }
+}
+
+impl TreeHash for KzgCommitment {
+    fn tree_hash_type() -> tree_hash::TreeHashType {
+        <[u8; BYTES_PER_COMMITMENT] as TreeHash>::tree_hash_type()
+    }
+
+    fn tree_hash_packed_encoding(&self) -> PackedEncoding {
+        self.0.tree_hash_packed_encoding()
+    }
+
+    fn tree_hash_packing_factor() -> usize {
+        <[u8; BYTES_PER_COMMITMENT] as TreeHash>::tree_hash_packing_factor()
+    }
+
+    fn tree_hash_root(&self) -> tree_hash::Hash256 {
+        self.0.tree_hash_root()
+    }
+}
+
+impl Serialize for KzgCommitment {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&format!("{self:?}"))
+    }
+}
+
+impl<'de> Deserialize<'de> for KzgCommitment {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let string = String::deserialize(deserializer)?;
+        Self::from_str(&string).map_err(serde::de::Error::custom)
+    }
+}
+
+impl FromStr for KzgCommitment {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Some(stripped) = s.strip_prefix("0x") {
+            let bytes = hex::decode(stripped).map_err(|e| e.to_string())?;
+            if bytes.len() == BYTES_PER_COMMITMENT {
+                let mut kzg_commitment_bytes = [0; BYTES_PER_COMMITMENT];
+                kzg_commitment_bytes[..].copy_from_slice(&bytes);
+                Ok(Self(kzg_commitment_bytes))
+            } else {
+                Err(format!(
+                    "InvalidByteLength: got {}, expected {}",
+                    bytes.len(),
+                    BYTES_PER_COMMITMENT
+                ))
+            }
+        } else {
+            Err("must start with 0x".to_string())
+        }
+    }
+}
+
+impl Debug for KzgCommitment {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", serde_utils::hex::encode(self.0))
+    }
+}
+
+#[test]
```

Comment:
I think we should continue the convention of using `mod tests {}` as a wrapper around the tests, which is how we typically do it across the code base.